### ISSUE TITLE
Append user hotstrings to separate file so that diffing is a bit easier?

### ISSUE THREAD

**Issue by Jason-K** [kunkel321/AutoCorrect2#1]
FileAppend("`n" WholeStr, A_ScriptFullPath)

==>

 FileAppend("`n" WholeStr, A_ScriptDir '\' UserHotstringsFile.ahk')?

**Comment by kunkel321** [comment]
Actually yeah,  That's a good idea.  I suppose we could also put an #Include HotstringsFile.ahk at the bottom of AutoCorect2.ahk.  A separate file might also make the validity scan (which currently scans the entire code) a tiny bit faster too though.  Thanks, I will make this change.

**Comment by kunkel321** [comment]
Hmmm...  On second thought...  I just tried putting the hotstrings in their own ahk file, then realized that the `f()` function definition, as well as Descolada's inputBuffer code have to be in the same file. 

Now I'm thinking:  Keep all that stuff in the `AutoCorrect2.ahk` but have `HotstringHelper2.ahk' in its own file.

**Comment by reza-repo** [comment]
100% agree with this idea👍

**Comment by Jason-K** [comment]
> Hmmm... On second thought... I just tried putting the hotstrings in their own ahk file, then realized that the `f()` function definition, as well as Descolada's inputBuffer code have to be in the same file.
> 
> EDIT: After a bit of experimentation, it looks like we can have the hotstrings in their own ahk file, and then #Include the file. This will allow us to access the f() function, but also will allow us to append to the bottom of the included hotstrings file.

Yeah - I've forked the repo and done some pretty substantial refactoring of the long functions, cleanup of tfhe global variable space, etc. I'll propose some pulls once I've finished testing. In the meantime, an #include at the end of the script should cover everything.

Separately, I've been having issues with short hotstrings. For example, :*?:his::f("his") consistently results in "iis"

**Comment by kunkel321** [comment]
Cool.  I'm looking forward to checking out what you've done.  Is there a way for me to view it now?  (I'm new to GitHub.)   Also, are the same person as Jasonosj, on the AHK forums?

**Comment by kunkel321** [comment]
Also....  Tell me more about the issue you are having?  I'm trying `:B0X:him::f("him")` and it seems to work fine.  In theory, it shouldn't do anything, really.  The strings are identical, so the function will determine that zero backspaces and the last zero characters of the replacement are need...  Are you sure you don't have mulitple instances of the exe running?

**Comment by kunkel321** [comment]
@Jason-K , I found my way to your fork.  _That's impressive work -- thank you for the work you've done on this!_  I definitely prefer your naming conventions.  Having all of the gui `OnEvent` commands separated seemed kindof strange at first, but I think I can get used to it.    What about the in-line comments though?   Can we put (some version of) those back?   I definitely prefer your version of the validation/check code too.  Like you said, the global variable space is quite a mess.  

You can probably tell that I'm not a programmer by trade or by training--LOL.

**Comment by Jason-K** [comment]
@kunkel321 Awesome! I've worked on it consistently and have been more focused on getting something that doesn't occasionally cause errors than integration with your code. Most of my changes are the result of my trying to debug and segregate things as I worked through issues in the debugger (e.g., I found it easier to look at the GUI function assignments as a block as opposed to interspersed throughout the GUI constructor, changed the order temporarily, moved on, never changed it back.)  For someone (like me) who doesn't have a background or training in coding, we're muddling our way through pretty well. Admittedly, we stand on the backs of competent contributors (leads?) like Descolada. 
One of the thoughts that I have been toying with has been storing the results of the function calls in replacement-specific map objects or a class with static variables. I think we can avoid a lot of duplicative logic if we, for example, calculate the replacement length once at the beginning and then call the result of that logic when it is needed later. 
Separately, have you experienced the caret being moved to the left and up a line after the hotstring send operation is completed? I don't know if the issue is the buffer, the script, or some interaction with one of the other scripts or apps I have running in the background (e.g., Ditto, Lightkey.io, etc.). I haven't had time to explore it and am curious if I'm the only one (suggests an issue outside the script).
